### PR TITLE
Use liter version of @repeaterjs/repeater

### PR DESCRIPTION
**Pull request by maneetgoyal**
https://github.com/gadget-inc/js-clients/assets/25290212/a85b2752-d9d8-4145-a50d-756834f33665

## PR Checklist

- [ ] Important or complicated code is tested
- [ ] Any user facing changes are documented in the Gadget-side changelog
- [ ] Any immediate changes are slated for release in Gadget via a generated package dependency bump
- [ ] Versions within this monorepo are matching and there's a valid upgrade path

### CODE REVIEW

**Comment by maneetgoyal** [comment]
@airhorns 

Added some tests around the new version of Repeater but getting a 1 failed test:

```
  ● importing from different contexts › can be imported from esm code

    expect(received).toBe(expected) // Object.is equality

    Expected: 0
    Received: 1

      12 |     const { status } = spawnSync("node", [fileURLToPath(new URL("named-esm-import.mjs", import.meta.url))]);
      13 |
    > 14 |     expect(status).toBe(0);
         |                    ^
      15 |   });
      16 | });
      17 |

      at Object.toBe (spec/imports/imports.spec.ts:14:20)
```

Debugging. Will try to add custom tests too thereafter.

**Comment by maneetgoyal** [comment]
> @airhorns
> 
> Added some tests around the new version of Repeater but getting a 1 failed test:
> 
> ```
>   ● importing from different contexts › can be imported from esm code
> 
>     expect(received).toBe(expected) // Object.is equality
> 
>     Expected: 0
>     Received: 1
> 
>       12 |     const { status } = spawnSync("node", [fileURLToPath(new URL("named-esm-import.mjs", import.meta.url))]);
>       13 |
>     > 14 |     expect(status).toBe(0);
>          |                    ^
>       15 |   });
>       16 | });
>       17 |
> 
>       at Object.toBe (spec/imports/imports.spec.ts:14:20)
> ```
> 
> Debugging. Will try to add custom tests too thereafter.

Failed test fixed.

File: packages/api-client-core/spec/Repeater.spec.ts [hunk 1]
```diff
@@ -0,0 +1,1577 @@
...
+    Repeater,
+    RepeaterOverflowError,
+    DroppingBuffer,
+    FixedBuffer,
+    MAX_QUEUE_LENGTH,
+    SlidingBuffer,
+} from "@manufac/repeater";
+import { delayPromise } from "./helpers.js";
+
+describe("Repeater", () => {
```
Comment (airhorns) [inline packages/api-client-core/spec/Repeater.spec.ts:11]: I think these tests should be in the forked package, not here, no? Either we should vendor both the code and the tests, or neither

File: packages/api-client-core/spec/applyLiveQueryJSONDiffPatch.spec.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        },
+    };
+
+    const stream = applyLiveQueryJSONDiffPatch(source);
+
+    await stream.next();
+    await stream.return();
+
+    expect(isCalled).toEqual(true);
+});
```
[the remark this answers is not in the document]
Comment (maneetgoyal) [reply packages/api-client-core/spec/applyLiveQueryJSONDiffPatch.spec.ts:160]: @airhorns Added a test for erroring.

File: packages/api-client-core/spec/Repeater.spec.ts (hunk 1 above)
Comment (maneetgoyal) [reply packages/api-client-core/spec/Repeater.spec.ts]: @airhorns Removed the tests completely. Once the PR is merged, I plan to migrate the forked Repeater package to this repo itself.

File: packages/api-client-core/spec/applyLiveQueryJSONDiffPatch.spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (maneetgoyal) [reply packages/api-client-core/spec/applyLiveQueryJSONDiffPatch.spec.ts:160]: @airhorns Should I add more tests?

**Review by airhorns — APPROVED** [review]

**Comment by airhorns** [comment]
@maneetgoyal it'd be great if you could open one more PR moving the fork into this package, thanks!

**Pull request merged**